src/components/pages/farmer-member/show/index.tsx

```diff
@@ -30,19 +30,18 @@ interface IFarmerShowProps {
 
 export default function FarmerShowPageComponent({ show }: { show: IFarmerShowProps }) {
   const router = useRouter();
-  const { user, previousPath } = useGlobalState();
-  const { backButtonText, backLink } = generateBackBtnStr(previousPath);
+  const { user, previousPath, setPreviousPath } = useGlobalState();
+  const { backButtonText, backLink } = generateBackBtnStr(previousPath, "Back to Farmer List");
+
+  if (!previousPath) {
+    setPreviousPath("/farmer/list");
```

Comment:
if there is no previous path, we'll redirect to farmer list?

Reply:
this is for new tab scenario.